File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/2095-minimum-number-of-swaps-to-make-the-string-balanced.swift
class Solution {
    func minSwaps(_ s: String) -> Int {
        var stackSize = 0;
        let n = s.count;

        for ch in s{
            if ch == "["{
                stackSize += 1;
            } else{
                if stackSize > 0{
                    stackSize -= 1;
                }
            }
        }
        return (stackSize + 1)/2
    }
}

File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/2095-minimum-number-of-swaps-to-make-the-string-balanced.rs
impl Solution {
    pub fn min_swaps(s: String) -> i32 {
        let mut stack_size = 0;
        let n = s.len();
        
        for ch in s.chars() {
            if ch == '[' {
                stack_size += 1;
            }
            else {
                if stack_size > 0 {
                    stack_size -= 1;
                }
            }
        }
        (stack_size + 1) / 2
    }
}

